Primo.js

```diff
@@ -318,6 +382,9 @@ function importPNX(text) {
 	item.edition = ZU.xpathText(doc, '//display/edition');
 	
 	var subjects = ZU.xpath(doc, '//search/subject');
+	if(!subjects.length) {
+		subjects = ZU.xpath(doc, '//display/subject');
+	}
```

Comment:
I'm working on integrating some of this -- decided against this after testing, since '//display/subject' isn't separated into tags, so we'd have to do this with a split, which carries a risk of error.